core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleManager.scala

```diff
@@ -156,10 +162,11 @@ private[spark] class SortShuffleManager(conf: SparkConf) extends ShuffleManager
 
   /** Remove a shuffle's metadata from the ShuffleManager. */
   override def unregisterShuffle(shuffleId: Int): Boolean = {
```

Comment:
I think we should ask for the shuffle generation id when `unregisterShuffle` is called.

Reply:
Maybe shuffle id is enough? Because only indeterminate stage has generation id and it should be the whole stage retried by same generation id, we have kept the tuple of the map number and the generation id in infoMapsForShuffle.